### ISSUE TITLE
Rust Support

### ISSUE THREAD

**Issue by meawoppl** [intrepidcs/libicsneo#69]
Hey all,

Great library. I have implemented a nearly complete Rust wrapper for the C api, and I was curious if you all had any interest in hosting it here. In some fashion.  I only have one device to test against, so my coverage is likely woefully low in some senses.
Do you all have some form of HIL testing you do to keep this library working?

In parallel, I have discovered a large number of (small) docs issues that I will contribute in either case. Let me know on the above, and thanks again for maintaining this.

**Comment by meawoppl** [comment]
Lovely...is there a way to use a .so with these calls instead? I don't see it in the build artifacts at time of writing.

**Comment by drebbe-intrepid** [comment]
@meawoppl in your cmake build directory if you are using `iicsneo_api` branch you'll have a `libicsneo.so` and `libicsneo-static.a` generated. The name is still pending and there are things that are broken.

**Comment by drebbe-intrepid** [comment]
@meawoppl I've completed a basic proof of concept for this new API and have a "functional" test cargo project here: https://github.com/intrepidcs/libicsneo/blob/icsneo_api/bindings/rust/icsneors/src/lib.rs

This is only binding generation and all unsafe currently. The "safe" layer still needs to be done and we are open to any feedback or recommendations including the work you have already done if you'd like to share it.
